posthog/api/alert.py

```diff
@@ -146,6 +157,7 @@ def create(self, validated_data: dict) -> AlertConfiguration:
         validated_data["created_by"] = self.context["request"].user
         subscribed_users = validated_data.pop("subscribed_users")
         threshold_data = validated_data.pop("threshold", None)
+        notification_destinations_data = validated_data.pop("notification_destinations", [])
```

Comment:
have you got tests to validate that this is not allowing someone to add a hog function from a different account? You don't seem to be doing any validation here of ownership which is pretty nasty 😬 

In addition, I think it would be better (unless there's a reason why it doesn't work) to simply create the hog function separately. So you load the alert and then load hog functions that have filters matching that alert trigger. Keeps everything decoupled and keeps all the existing permissions working as they should.

It might be a little complex if you haven't worked with the api much but I can explain it and see if it works for your case?

Reply:
Aah I see what you mean! Surveys/actions are a good example.
I'll remove the db migration and just use event filters to match destinations to alerts + still have templates for insight alert firing